hydroflow/src/builder/hydroflow_builder.rs

```diff
@@ -85,12 +87,20 @@ impl HydroflowBuilder {
     ) -> SubgraphId
     where
         Name: Into<Cow<'static, str>>,
-        Pull: 'static + PullSurface,
-        Push: 'static + PushSurfaceReversed<ItemIn = Pull::ItemOut>,
+        Pull: 'static + PullSurface + TrackPullDependencies,
+        Push: 'static + PushSurfaceReversed<ItemIn = Pull::ItemOut> + TrackPushDependencies,
     {
+        let mut deps = DirectedEdgeSet::default();
```

Comment:
nit: i think this block of code would be good to extract into a helper

Reply:
replaced with `DirectedEdgeSet::new()`